### PR TITLE
Adds Details for Collection Release

### LINKED ISSUE #769: Display more details on collection view page for delayed release
In the release and visibility of information section on the collection view page, one of the following should display. When the collection manager chooses 

1. immediate release, display "Immediately"
2. to delay release until a specific date for all deposits, display "[months/years] from date of deposit"
3. to allow the depositor to choose, "depositor selects release date at no more than x [months/years] from date of deposit"

### PR DESCRIPTION
**Pull request by jermnelson**
## Why was this change made?
Fixes #769 

Depending on the collection manager's choices for Release, the following messages are displayed.

Immediately:
Screen Shot 2021-02-18 at 10 53 59 AM

With a specific delayed time period:
Screen Shot 2021-02-18 at 10 53 44 AM

When depositor selects a date:
Screen Shot 2021-02-18 at 10 54 12 AM

## How was this change tested?
New component tests.

## Which documentation and/or configurations were updated?
n/a

### CODE REVIEW

File: app/components/collections/release_component.rb [hunk 1]
```diff
@@ -12,6 +12,18 @@ def initialize(collection:)
     sig { returns(Collection) }
     attr_reader :collection
 
-    delegate :release_option, :release_duration, :access, to: :collection
```
Comment (mjgiarlo) [inline app/components/collections/release_component.rb:15]: Can you say more about this change?

**Review by mjgiarlo — APPROVED** [review]
Minor question about a change to delegation in the component but LGTM otherwise

**Review by mjgiarlo — APPROVED** [review]
Minor question about a change to delegation in the component but LGTM otherwise

**Review by mjgiarlo — APPROVED** [review]
Minor question about a change to delegation in the component but LGTM otherwise

File: app/components/collections/release_component.rb (hunk 1 above)
Comment (jermnelson) [reply app/components/collections/release_component.rb:15]: We're no longer using `release_option` or `release_duration` in the component template, replaced with new `release_info`. Happy to restore if it is used elsewhere.

Comment (mjgiarlo) [reply app/components/collections/release_component.rb:15]: OK, that makes sense then not to expose them in the public interface. Good call, thanks!

**Review by mjgiarlo — APPROVED** [review]

**Pull request merged**